helper-cli/src/main/kotlin/commands/packagecuration/SplitCommand.kt

```diff
@@ -57,38 +56,38 @@ internal class SplitCommand : CliktCommand(
         .required()
 
     override fun run() {
-        if (!outputCurationsDir.list().isNullOrEmpty()) {
-            throw UsageError("Output directory $outputCurationsDir must be empty.")
-        }
-
         val packageCurations = inputCurationsFile.readValue<List<PackageCuration>>()
         val groupedCurations = packageCurations.groupBy {
             getSplitCurationFile(outputCurationsDir, it.id, inputCurationsFile.extension)
         }
 
         val mapper = createBlockYamlMapper()
         groupedCurations.forEach { (outputFile, curations) ->
-            curations.forEach { originalCuration ->
+            val curationsToPersist = if (outputFile.isFile) {
+                outputFile.readValue<MutableSet<PackageCuration>>()
+            } else {
+                mutableSetOf()
+            }
+
+            curationsToPersist += curations
+
+            val curationsWithBlockComment = curationsToPersist.map { originalCuration ->
                 val comment = originalCuration.data.comment?.wrapAt(COMMENT_WRAP_COLUMN)
 
-                val curation = if (comment != null) {
+                if (comment != null) {
                     // Ensure at least a single "\n" is contained in the comment to force the YAML mapper to use block
                     // quotes.
                     originalCuration.copy(data = originalCuration.data.copy(comment = "$comment\n"))
                 } else {
                     originalCuration
                 }
+            }.toSet()
 
-                val text = mapper.writerWithDefaultPrettyPrinter().writeValueAsString(listOf(curation))
+            val text = mapper.writerWithDefaultPrettyPrinter()
+                .writeValueAsString(curationsWithBlockComment.sortedBy { it.id.version })
```

Comment:
Just a thought, not sure if it's nicer: You could put this before line 64

    val curationsWithBlockComment = sortedSetOf(compareBy<PackageCuration> { it.id.version})

and use in  line 74

    curationsToPersist.mapTo(curationsWithBlockComment) { originalCuration ->

to get rid of `toSet()` in line 84 and of `sortedBy {}` in line 87. In exchange you'd need to add

    curationsWithBlockComment.clear()

in line 91.

Reply:
In my opinion this would make the code less readable.